resources/ext/config/conf.d/puppetserver.conf

```diff
@@ -1,14 +1,20 @@
+# Where the puppet-agent dependency places puppet, facter, etc...
+# Puppet server expects to load Puppet from this location
+os-settings: {
+    ruby-load-path: [/opt/puppetlabs/puppet/lib/ruby/vendor_ruby]
+}
+
 # configuration for the JRuby interpreters
 jruby-puppet: {
     # This setting determines where JRuby will look for gems.  It is also
     # used by the `puppetserver gem` command line tool.
-    gem-home: /var/lib/puppet/jruby-gems
+    gem-home: /opt/puppetlabs/server/data/puppetserver/jruby-gems
 
     # (optional) path to puppet conf dir; if not specified, will use the puppet default
-    master-conf-dir: /etc/puppetlabs/agent
+    master-conf-dir: /etc/puppetlabs/puppet
 
     # (optional) path to puppet var dir; if not specified, will use the puppet default
-    master-var-dir: /opt/puppetlabs/agent/cache
+    master-var-dir: /opt/puppetlabs/server/data/puppetserver
```

Comment:
It is actually critical that `puppet master --configprint vardir` return the "correct" value, because beaker uses that to start/stop puppetserver during acceptance tests: https://github.com/puppetlabs/beaker/blob/master/lib/beaker/dsl/helpers.rb#L563

So I think it's ok to split the vardirs in this PR, but the postinstall step that executes `/opt/puppetlabs/puppet/bin/puppet config set --section master vardir  /opt/puppetlabs/server/data/puppetserver` must remain also.

In the future, if you decide to remove the postinstall step, please don't add another beaker configuration option, e.g. `puppetserver-vardir`, because that tightly couples beaker and puppet-server. I think it would be preferable if there was a way beaker could interrogate the host to determine this, along the lines of `puppetserver config print vardir`. That could also be used to retrieve `confdir` and then we can deprecate `puppetserver-confdir` from beaker and node definition files.